web/packages/shared/deepLinks.ts

```diff
@@ -106,6 +105,8 @@ export function makeDeepLinkWithSafeInput<
     : '';
 
   const searchParamsString = Object.entries(args.searchParams)
+    // filter out params that have no value to prevent a string like "&myparam=null"
+    .filter(kv => !!kv[1])
```

Comment:
That will filter out `0`, `false` etc as well.
I'd check for `null` and `undefined` explicitly here.